src/TriggerBinding/SqlTriggerBinding.cs

```diff
@@ -0,0 +1,96 @@
+        {
+            this._tableName = tableName ?? throw new ArgumentNullException(nameof(tableName));
+            this._connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+            this._parameter = parameter ?? throw new ArgumentNullException(nameof(parameter));
+            this._hostIdProvider = hostIdProvider ?? throw new ArgumentNullException(nameof(hostIdProvider));
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        /// Returns the type of trigger value that <see cref="SqlTriggerBinding{T}" /> binds to.
+        /// </summary>
+        public Type TriggerValueType => typeof(IReadOnlyList<SqlChange<T>>);
+
+        public IReadOnlyDictionary<string, Type> BindingDataContract => _emptyBindingContract;
+
+        public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
+        {
+            IValueProvider valueProvider = new SqlTriggerValueProvider(this._parameter.ParameterType, value, this._tableName);
+            return Task.FromResult<ITriggerData>(new TriggerData(valueProvider, _emptyBindingData));
+        }
+
+        public async Task<IListener> CreateListenerAsync(ListenerFactoryContext context)
+        {
+            _ = context ?? throw new ArgumentNullException(nameof(context), "Missing listener context");
+
+            string userFunctionId = await this.GetUserFunctionIdAsync();
+            return new SqlTriggerListener<T>(this._connectionString, this._tableName, userFunctionId, context.Executor, this._logger);
+        }
+
+        public ParameterDescriptor ToParameterDescriptor()
+        {
+            return new SqlTriggerParameterDescriptor
+            {
+                Name = this._parameter.Name,
+                Type = "SqlTrigger",
+                TableName = this._tableName,
+            };
+        }
+
+        /// <summary>
+        /// Returns an ID that uniquely identifies the user function.
+        ///
+        /// We call the WebJobs SDK library method to generate the host ID. The host ID is essentially a hash of the
+        /// assembly name containing the user function(s). This ensures that if the user ever updates their application,
+        /// unless the assembly name is modified, the new application version will be able to resume from the point
+        /// where the previous version had left. Appending another hash of class+method in here ensures that if there
+        /// are multiple user functions within the same process and tracking the same SQL table, then each one of them
+        /// gets a separate view of the table changes.
+        /// </summary>
+        private async Task<string> GetUserFunctionIdAsync()
+        {
+            string hostId = await this._hostIdProvider.GetHostIdAsync(CancellationToken.None);
```

Comment:
Are there any public docs on this function? I don't see any here : https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.webjobs.host?view=azure-dotnet

But from your comment and looking at https://github.com/Azure/azure-webjobs-sdk/tree/dev/src/Microsoft.Azure.WebJobs.Host/Executors - does this mean that if a user deploys a the same function (with trigger binding) to two separate function apps that the host ID will be the same? Which means they'd share the same worker tables and state - is that the intended behavior?

Reply:
This is the best effort approach. The class behind `IHostProvider` is `DefaultHostIdProvider`. As can be seen in [its implementation](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host/Executors/DefaultHostIdProvider.cs#L56) hashes the name of the assembly that contains user functions. This ensures that if user modifies and rebuilds their application, until the assembly's name is not updated, the worker table and global state is reused. If this is not a plane user application but a function apps, the functions host overwrites the `DefaultHostIdProvider` with its [own implementation](https://github.com/Azure/azure-functions-host/blob/bfecdc29f78cd29cb6a8d01181d9f24ca79fe3a8/src/WebJobs.Script/Host/ScriptHostIdProvider.cs) which takes into account more parameters and updates the parameter set based on where the function app is hosted (on Azure or inside Kubernetes cluster).

Reply:
Good to know about the web host ID, thanks.

There's a couple concerns/questions I have : 

1. If a user wants to have the trigger start "fresh" (picking up events from the current state going forward) then they have to make sure to clear out existing worker tables/global state. But it's not clear how they would know what to do - the IDs used don't really help indicate which one belongs to which instance so the user might end up deleting something they didn't mean to. 
2. Trigger instances might end up sharing state that isn't intended. If I have two applications running that are essentially the same but let's say have a different config controlling what they do with the events then they'll end up sharing an ID, right? That doesn't seem ideal since there isn't any way to work around that. Maybe not a common scenario but what if we at least documented a config value that could be set to manually control the ID in case the user wants to either guarantee uniqueness (or specifically re-use the same state)

How do existing triggers work - are they similar in that they try to keep the state for the trigger across multiple instances? I would have initially thought that the SQL trigger would always just start listening for events as soon as it started - not caring about picking up past events or anything. Is all this state logic here mostly meant for handling scaling (where my understanding is that there might be multiple instances of the same worker running in parallel) or is there other functionality that this is intended for?

Reply:
Once we decide on what the correct behavior is make sure to document this clearly in the public docs as well - users need to know: 

1. In what specific cases state will be shared - be as explicit as possible with how this is determined
2. How to clean up state for an instance manually

Reply:
Tracking this in #283